src/metabase/util.cljc

```diff
@@ -1208,6 +1208,19 @@
   [s]
   (when-not (str/blank? s) s))
 
+(defn safe-min
+  "nil safe clojure.core/min"
+  [& args]
+  (transduce
+   (filter some?)
+   (completing
+    (fn [acc n]
+      (if acc
+        (min acc n)
+        n)))
+   nil
+   args))
```

Comment:
Could you add a quick unit test for this? The implementation is non-trivial so I think it's worth validating it.

Not totally sure why it's written like this either. I guess it's marginally faster for large amounts of data since it doesn't build up any intermediate data structures.

Reply:
Here's a quick profiling comparison with a simpler implementation:
```
(defn- safe-min2
  [& args]
  (when-let [filtered (seq (remove nil? args))]
    (apply min filtered)))

(comment
  (require '[criterium.core :as c])
  (def test-data (concat (repeat 50000 nil) (range 100000)))
  (c/quick-bench (apply safe-min test-data))
  (c/quick-bench (apply safe-min2 test-data)))
```

`safe-min` has an average execution time of 2.275417 ms and `safe-min2` has an average of 2.808365 ms. 🤷